src/python/platforms/android/adb.py

```diff
@@ -419,11 +435,31 @@ def reset_device_connection():
   state = get_device_state()
   if state != 'device':
     logs.log_warn('Device state is %s, unable to recover using usb reset/'
-                  'gce reconnect.' % str(state))
+                  'cuttlefish reconnect.' % str(state))
     return False
 
   return True
 
+def get_cuttlefish_device_ip():
+  try:
+    return socket.gethostbyname('cuttlefish')
+  except socket.gaierror:
+    logs.log_warn('Failed to get cuttlefish ip address, use localhost instead')
+    return '127.0.0.1'
+
+
+def set_cuttlefish_device_serial():
+  device_serial = '%s:%d' % (get_cuttlefish_device_ip(), CUTTLEFISH_CVD_PORT)
+  environment.set_value('ANDROID_SERIAL', device_serial)
+
+
+def get_cuttlefish_address():
+  ip = get_cuttlefish_device_ip()
+  if ip == '127.0.0.1':
+      logs.log_fatal_and_exit(
```

Comment:
If `127.0.0.1` is expected to be an error, then we can just do this in `get_cuttlefish_device_ip` instead, rather than warning and returning it if we're going to end up calling log_fatal anyway.